Lib/importlib/_bootstrap.py

```diff
@@ -54,48 +54,180 @@ def _new_module(name):
 # A dict mapping module names to weakrefs of _ModuleLock instances
 # Dictionary protected by the global import lock
 _module_locks = {}
-# A dict mapping thread ids to _ModuleLock instances
+
+# A dict mapping thread ids to lists of _ModuleLock instances.  This maps a
+# thread to the module locks it is blocking on acquiring.  The values are
+# lists because a single thread could perform a re-entrant import and be "in
+# the process" of blocking on locks for more than one module.  "in the
+# process" because a thread cannot actually block on acquiring more than one
+# lock but it can have set up bookkeeping that reflects that it intends to
+# block on acquiring more than one lock.
 _blocking_on = {}
 
 
+class _BlockingOnManager:
+    """
+    A context manager responsible to updating ``_blocking_on`` to track which
+    threads are likely blocked on taking the import locks for which modules.
+    """
+    def __init__(self, tid, lock):
+        # The id of the thread in which this manager is being used.
+        self.tid = tid
```

Comment:
```suggestion
    def __init__(self, thread_id, lock):
        self.thread_id = thread_id
```

I realize there are other places that use `tid` for a variable name, but I don't think that's descriptive enough on its own.